src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerWithContextGenerator.cs

```diff
@@ -87,13 +87,15 @@ public class ControllerWithContextGenerator : ControllerGeneratorBase
                 //Todo: Pluralize model name
                 controllerGeneratorModel.ControllerName = modelTypeAndContextModel.ModelType.Name + Constants.ControllerSuffix;
             }
-
+            var namespaceName = string.IsNullOrEmpty(controllerGeneratorModel.ControllerNamespace)
```

Comment:
Need to spend some time to come up with end to end tests here. 
Currently end to end tests are in the Visual Studio code base.